src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,42 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int [] result = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            result[i] = matrix[i][i];
+        }
+        return result;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] result = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            result[i] = matrix[i][matrix[i].length - 1 - i];
+        }
+        return result;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+        for (int[] ints : matrix) {
+            for (int j = 0; j < matrix.length; j++) {
```

Comment:
let's use two for loop instead of foreach

Reply:
Done.
But, can you explain why we need to use for loop instead of foreach?